---
title: "Visualize gene expression"
author: "Guillem Ylla"
date: "2023-08-03"
output: html_document
---

To run a single line where you have the cursor:

- Ctrl/Command + Enter

To run all current chunk:

- Ctrl/Command + Shift + Enter 

## Load data

```{r}
# Load table of counts
Table_of_counts<-read.csv("data/GSE99785_gene_Counts_table_anot_v3.csv",row.names=1)
# Visualize first rows
head(Table_of_counts)
```

Check the library sizes:
 
```{r}
colSums(Table_of_counts)
```


## Normalize in TPMs

Transcript per Million (TPM) normalization: normalizes the counts by gene length and library size. We will divide the counts of each gene, by the gene length, and then, by the library size.


- Get gene lengths:

```{r}
# Load gene length table. It can be obtained from gff/gtf file. Here we have it prepared.
gene_len<-read.csv("data/Bger_gene_legth.csv")
```

- Divide the counts by the gene length:

```{r}
# Let's make sure that the genes in the table of counts and in the gene length table are in same order
identical(rownames(Table_of_counts), gene_len$GeneID)

# Calculate reads per kb (dividing the counts by the length of the gene per kb)
RPK=(Table_of_counts/ (gene_len$Length/1000))
#RPK[is.na(RPK)] <- 0
# Calculate library size (sum of RPK) and divide per 1M
RPK_sum=colSums(RPK)/1000000
# Get TPMs by dividing the RPK by the library size
TPM=sweep(RPK, MARGIN=2, STATS=RPK_sum, FUN="/", check.margin=TRUE) # MARGIN=2 applies the FUNction (/) to each column of RPK using the STATS object
# Visualize first rows of the TPM table
head(TPM)
```


```{r}
# The sum of TPMs, by definition should be 1M in all samples
colSums(TPM)
```

## Select a gene of interest

```{r}
ID_gene_of_interest<-"Bger_05466"

GoI_TPM<-TPM[ rownames(TPM) == ID_gene_of_interest, ]

GoI_TPM
```

## Barplot of the gene

```{r}
barplot(as.matrix(GoI_TPM), col="darkgrey")
```

----

## Advanced exercises

### Make nicer plots

With ggplot2 you can make much nicer plots!

```{r message=FALSE, warning=FALSE}
#install.packages(c("tidyr","ggplot2","dplyr","tidyr","tibble","stringr")) # Uncoment the line and run if you need to install the libraries
library(ggplot2)
library(dplyr)
library(tidyr)
library(tibble)
library(stringr)

# for ggplots we first need to transfrom the wide table into a long table
TPM_long<-TPM %>% 
  rownames_to_column("GeneIDs") %>% 
  pivot_longer(cols=colnames(TPM),names_to="Sample", values_to = "TPMs") %>%
  separate(Sample, sep="_", into= c("Stage","Replicate"), remove = FALSE ) 

TPM_long_GoI<-TPM_long %>%  
                filter(GeneIDs=="Bger_05466")# select TPMs of the gene of interest
  

ggplot(TPM_long_GoI, aes(x=Stage, y=TPMs))+
  geom_point(aes(color=Stage), size=4)+ # plot data in points and color by stage
  ggtitle(paste("Expression of the gene - ",TPM_long_GoI$GeneIDs)) + # add title
  theme_light() # use a pre-defined light theme
```

### Plot multiple genes at ones

```{r}


TPM_long_multiple_genes<-TPM_long %>%  
                filter(GeneIDs%in%c("Bger_05466","Bger_05260","Bger_03560","Bger_03657")) ## select multiple genes
  

ggplot(TPM_long_multiple_genes, aes(x=Stage, y=TPMs))+
  geom_point(aes(color=Stage), size=4)+
  facet_wrap(~GeneIDs, ncol = 2)+ # separate plots by GeneIDs in 2 columns
  ggtitle("Expression of the selected genes") +
  theme_light()
```


