---
title: "Visualize gene expression"
author: "Guillem Ylla"
date: "2023-07-06"
output: html_document
---

```{r setup, include=FALSE}
## Set parameters
knitr::opts_chunk$set(echo = TRUE)
```

## Load data

```{r}
#load table of counts
Table_of_counts<-read.csv("GSE99785_gene_Counts_table_anot_v3.csv",row.names=1)

head(Table_of_counts)
```
## Normalize in TPMs

Transcript per Million (TPM) normalization, normalized by gene length and library size. We will divide the counts of each gene, by the gene length, and then, by the total number of reads in the sample (i.e. library size).


- Get gene lengths:

```{r}
# Load gene length table. It can be obtained from gff/gtf file. Here we have it prepared.
gene_len<-read.csv("Bger_gene_legth.csv")
```

- Devide the counts by the gene length:

```{r}
# Cet's make sure that the genes in the table of counts and in the gene length table are in same order
identical(rownames(Table_of_counts), gene_len$GeneID)

# Calculate reads per kilobase (dividing the counts by the length of the gene per kb)
RPK=(Table_of_counts/ (gene_len$Length/1000))
#RPK[is.na(RPK)] <- 0
# Calculate library size (sum of RPK) and divide per 1M
RPK_sum=colSums(RPK)/1000000
# Get TPMs by dividing the RPK by the library size
TPM=sweep(RPK, MARGIN=2, STATS=RPK_sum, FUN="/", check.margin=TRUE) # MARGIN=2 applies the FUNction (/) to each column of RPK using the STATS object
# see first rows of the TPM table
head(TPM)
```


```{r}
# The sum of TPMs, by definition should be 1M in all samples
colSums(TPM)
```

## Get Gene of Interest

```{r}
ID_gene_of_interest<-"Bger_05466"

GoI_TPM<-TPM[ rownames(TPM) == ID_gene_of_interest, ]

GoI_TPM
```

## Barplot of the gene

```{r}
barplot(as.matrix(GoI_TPM), col="darkgrey")

```


